lib/linked_list.rb

```diff
@@ -94,19 +151,44 @@ def has_cycle
     # returns the value in the first node
     # returns nil if the list is empty
     def get_first
-      raise NotImplementedError
+      if @head.nil? 
+        return nil 
+      else
+        return @head.data 
+      end
     end
 
     # method that inserts a given value as a new last node in the linked list
     def add_last(value)
-      raise NotImplementedError
+      if @head.nil? 
+        @head = Node.new(value, nil)
+        return
+      end
+      current = @head
+      until current.next.nil?
+        current = current.next
+      end
+      current.next = Node.new(value, nil) 
     end
 
+
     # method that returns the value of the last node in the linked list
     # returns nil if the linked list is empty
-    def get_last
-      raise NotImplementedError
+    def get_last(value)
+    previous = @head 
+    current = @head 
+
+    until current.nil?

```

Comment:
I'd suggest going `until current.next.nil?`, and then setting current.next equal to a new node.